src/test/java/co/cask/gcp/datastore/sink/DatastoreSinkTest.java

```diff
@@ -0,0 +1,545 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.gcp.datastore.sink;
+
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.api.dataset.lib.KeyValue;
+import co.cask.cdap.etl.api.Emitter;
+import co.cask.gcp.datastore.sink.util.IndexStrategy;
+import co.cask.gcp.datastore.sink.util.SinkKeyType;
+import com.google.cloud.Timestamp;
+import com.google.cloud.datastore.FullEntity;
+import com.google.cloud.datastore.IncompleteKey;
+import com.google.cloud.datastore.Key;
+import com.google.cloud.datastore.KeyFactory;
+import com.google.cloud.datastore.PathElement;
+import org.apache.hadoop.io.NullWritable;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.time.ZonedDateTime;
+import java.util.Collections;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * Tests for {@link DatastoreSink} class.
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class DatastoreSinkTest {
+
+  @Rule
+  public ExpectedException thrown = ExpectedException.none();
+
+  @InjectMocks
+  private DatastoreSink datastoreSink;
+
+  @Mock
+  private DatastoreSinkConfig config;
+
+  @Mock
+  private Emitter<KeyValue<NullWritable, FullEntity<?>>> emitter;
+
+
+  @SuppressWarnings("ConstantConditions")
+  @Test
+  public void testTransformAllTypesIndexStrategyAll() {
```

Comment:
these unit tests become simpler and easier to understand if the transform logic is moved to a different class.